Documentation/container-lifecycle.md

```diff
@@ -0,0 +1,34 @@
+# Lifecycle of a container in rocket
+
+## Creation
+
+`rkt run` instantiates a container uuid and filesystem tree rooted at `/var/lib/rkt/containers/$uuid/.`, henceforth referred to as `$cdir`.
+
+An exclusive advisory lock is acquired on `$cdir` with a file descriptor associated with the process executing the stage1 /init.  This lock is automatically released when the /init process exits and the kernel closes all of the remaining open file descriptors.
```

Comment:
Why is this lock needed? This is a good moment to teach someone how it works a little deeper.